src/Util.hs

```diff
@@ -22,3 +24,18 @@ normalizeTypeSignatures = \case
   xs | "\n  :: " `isPrefixOf` xs -> normalizeTypeSignatures (drop 2 xs)
   x : xs -> x : normalizeTypeSignatures xs
   [] -> []
+
+dotGhciWritableByOthers :: IO Bool
+dotGhciWritableByOthers = do
+  exists <- fileExist ".ghci"
+  if exists then do
+    mode <- fileMode <$> getFileStatus ".ghci"
+    dirMode <- fileMode <$> getFileStatus "."
+    return (writableByOthers mode || writableByOthers dirMode)
```

Comment:
I'm thinking it could be easier to read if we had `includesMode :: FileMode -> FileMode -> Bool` and use that four times here. (Well, currently it's actually `writableByOthers` that's hard to understand. But we would get rid of that.)